src/model/deck.rs

```diff
@@ -59,6 +61,10 @@ impl Deck {
         self.mainboard.insert_many(card, count)
     }
 
+    pub fn get_card(&self, card: &AbstractCard) -> AbstractCard {
```

Comment:
I doubt the usefulness of this method. Let's say the deck does contain the given card in the mainboard. You are getting it, cloning it, and returning it to the user. However, the returned card should contain all the same information that the given `card` does. Based on our conversation elsewhere, I believe you want to add methods on the `AbstractCard` struct.